### PR TITLE
Fix: Scale rating minigraphs on station window list.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

Minigraphs did not adjust size to accomodate large text, either by font size or font zoom, leading to cropped labels.

## Description

Minigraphs and spacing are now scaled by font zoom, as this seems to behave better than gui zoom in this instance.
Magic scaling values have been simplified and logic is set to explicitly treat cargo waiting <= 30 as a small vertical chart.

## Limitations

The pixel-doubling logic to make the rating bar more visible assumes the font size is increased by font zoom level, rather than by setting a larger font size.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by PeterN** [comment]
Before this change:
stationlistbefore
After this change:
stationlistafter

**Comment by PeterN** [comment]
Totally forgot but this change also fixes the sizing/spacing of the transport type / cargo filter buttons at the top of the window.

File: src/station_gui.cpp [hunk 1]
```diff
@@ -445,7 +445,14 @@ class CompanyStationsWindow : public Window
...
 				int y = r.top + WD_FRAMERECT_TOP;
-				for (int i = this->vscroll->GetPosition(); i < max; ++i) { // do until max number of stations of owner
+				uint line_height = this->GetWidget<NWidgetBase>(widget)->resize_y;
+				/* Spacing between station name and first rating graph. */
+				uint text_spacing = ScaleFontTrad(5);
+				/* Spacing between additional rating graphs. */
+				uint rating_spacing = ScaleFontTrad(4);
+
+				for (int i = this->vscroll->GetPosition(); i < max; ++i)
+				{ // do until max number of stations of owner
```
Comment (LordAro) [inline src/station_gui.cpp]:
```suggestion
				// do until max number of stations of owner
				for (int i = this->vscroll->GetPosition(); i < max; ++i) {
```

File: src/station_gui.cpp [hunk 2]
```diff
@@ -165,32 +162,33 @@ static void StationsWndShowStationRating(int left, int right, int y, CargoID typ
 	const CargoSpec *cs = CargoSpec::Get(type);
 	if (!cs->IsValid()) return;
 
+	int one = ScaleFontTrad(1);
```
Comment (LordAro) [inline src/station_gui.cpp]: perhaps "padding" instead of "one" ?

File: src/station_gui.cpp (hunk 1 above)
Comment (PeterN) [reply src/station_gui.cpp]: Have reverted this change, unintentional autoreformat!

**Review by LordAro — APPROVED** [review]

**Pull request merged**